src/DynamoCoreWpf/UI/BindingTraceListener.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Diagnostics;
+using System.Text;
+using System.Windows;
+
+namespace Dynamo.UI
+{
+    public class BindingErrorTraceListener : DefaultTraceListener
+    {
```

Comment:
Never put in code that you haven't written without clear attribution and check that it matches our licensing.  

https://github.com/StanleyGoldman/TeamPilgrim/blob/master/src/TeamPilgrim.VisualStudio/Common/BindingErrorTraceListener.cs

Also, from looking at the code I have no idea what this codes does.  It should have comments that clarify that.